Assets/SEE/Layout/NodeLayouts/IncrementalTreeMapLayout.cs

```diff
@@ -0,0 +1,354 @@
+            {
+                LocalMoves.DeleteNode(obsoleteNode);
+                workWith.Remove(obsoleteNode);
+            }
+
+            CorrectAreas.Correct(workWith, _settings);
+            foreach (var nodeToBeAdded in nodesToBeAdded)
+            {
+                LocalMoves.AddNode(workWith, nodeToBeAdded);
+                workWith.Add(nodeToBeAdded);
+            }
+
+            CorrectAreas.Correct(workWith, _settings);
+
+            LocalMoves.LocalMovesSearch(workWith, _settings);
+        }
+
+        /// <summary>
+        /// Calculates 3 lists of <see cref="Node"/>s to transform stepwise the old layout
+        /// to a new one.
+        /// </summary>
+        /// <param name="nodes">siblings from the current layout</param>
+        /// <param name="oldNodes">corresponding siblings from the old layout</param>
+        /// <param name="workWith">a copy of the old layout with nodes from the new one</param>
+        /// <param name="nodesToBeDeleted">artificial nodes with no equivalent ILayoutNode, part of workWith</param>
+        /// <param name="nodesToBeAdded">nodes that are not in workWith, but in nodes</param>
+        private void SetupNodeLists(
+            List<Node> nodes,
+            List<Node> oldNodes,
+            out List<Node> workWith,
+            out List<Node> nodesToBeDeleted,
+            out List<Node> nodesToBeAdded)
+        {
+            //  [         oldNodes            ]--------------   <- nodes of OLD layout
+            //  --------------[              nodes          ]   <- nodes of new layout
+            //  [ toBeDeleted ]---------------[  toBeAdded  ]   <- nodes of new layout
+            //  [         workWith            ]--------------   <- nodes of new layout
+            //                                                     designed to be changed over time to nodes
+
+            // get nodes form old layout .. copy their rectangles
+            // setup workWith and nodesToBeDeleted
+            workWith = new List<Node>();
+            nodesToBeDeleted = new List<Node>();
+            foreach (var oldNode in oldNodes)
+            {
+                Node newNode = nodes.Find(x => x.ID.Equals(oldNode.ID));
+                if (newNode == null)
+                {
+                    // create a artificial node, that has no corresponding ILayoutNode in this layout
+                    // they are designed to be deleted but necessary to copy the layout of oldNodes
+                    newNode = new Node(oldNode.ID);
+                    nodesToBeDeleted.Add(newNode);
+                }
+
+                workWith.Add(newNode);
+                newNode.Rectangle = oldNode.Rectangle.Clone();
+            }
+
+            IncrementalTreeMap.Utils.CloneSegments(
+                from: oldNodes,
+                to: workWith.ToDictionary(n => n.ID, n => n));
+
+            var workWithAlias = workWith;
+            nodesToBeAdded = nodes.Where(n => !workWithAlias.Contains(n)).ToList();
```

Comment:
Why is this alias defined? `workWith` is not modified, so it should be fine to use it directly, right?

Reply:
Because ```workWith``` is a ```out``` parameter and cant be used in lambda expressions.
This was my silly work around, exists here any better solution?

Reply:
Ah, I didn't know you cannot use `out` parameters in lambdas. This workaround is fine.